### ISSUE TITLE
Issue with installing packages from private conda repositories in version `0.11.0`

### ISSUE THREAD

**Issue by mohahf19** [prefix-dev/pixi#631]
### Checks

- [X] I have checked that this issue has not already been reported. (similar to https://github.com/prefix-dev/pixi/issues/334)

- [X] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

on ubuntu 22.04: 

```shell
pixi --version 
# pixi 0.11.0
pixi auth login --conda-token my-token https://my-url -v
# WARN rattler_networking::authentication_storage::storage: Error storing credentials in backend: 
# Platform secure storage failure: no secret service provider or dbus session found
```
Checking the credentials file (`/root/.rattler/credentials.json`), I see:

```
{"my-url":{"CondaToken":"my-token"}
``` 

However running `pixi install` fails due to a 401 authentication error. 

On the other hand, on `pixi 0.10.0`, 
```shell
pixi --version
# pixi 0.10.0
pixi auth login --conda-token my-token https://my-url -v
Authenticating with my-url
# WARN rattler_networking::authentication_storage::storage: Error storing credentials for my-url: 
# Platform secure storage failure: no secret service provider or dbus session found, 
# using fallback storage at /root/.rattler/rattler_auth_store.json
# WARN rattler_networking::authentication_storage::fallback_storage: Can't find path for 
# fallback storage on /root/.rattler/rattler_auth_store.json
```
but the token is in the `/root/.rattler/rattler_auth_store.json` file. and then running `pixi install` works. 

### Issue description

Since upgrading to version`0.11.0`, I am having issues with authenticating with private conda repositories and installing packages. From what I see, version `0.11.0` on ubuntu 22.04, is able to store the token in the credentials file (not the system store). However when I try to run `pixi install`, I get a 401 authentication error. 

### Expected behavior

I would expect `pixi` to be able to be able to authenticate with a private repository even if it could not store the credentials in the backend by using the file fall back (version `0.10.0` behavior), and to be able to install packages from it.

**Comment by wolfv** [comment]
Maybe @0xbe7a has an idea, too :)

**Comment by 0xbe7a** [comment]
rattler only checks fallback stores if all previous stores have returned errors.

https://github.com/mamba-org/rattler/blob/main/crates/rattler_networking/src/authentication_storage/storage.rs#L95

If the keyring store encounters an error, it returns `Ok(None)` instead of Error, signaling that the search was successful but no credentials were found. Instead, it should examine the error that occurred and only return `Ok(None)` for `NoEntry`.

https://github.com/mamba-org/rattler/blob/main/crates/rattler_networking/src/authentication_storage/backends/keyring.rs#L67

**Comment by wolfv** [comment]
Thanks @0xbe7a for the fix!

**Comment by pavelzw** [comment]
@0xbe7a is this the same issue as https://github.com/prefix-dev/setup-pixi/pull/64? It would be nice to get a patched release 0.11.1 that fixes the issue :)

**Comment by ruben-arts** [comment]
@pavelzw CI is crunching it out right now ;)
